modules/internal/CString.chpl

```diff
@@ -120,13 +120,29 @@ module CString {
     return __primitive("cast", t, x);
   }
   //
+  // casts from c_string to c_ptrConst(c_char/int(8)/uint(8))
+  //
+  inline operator :(x: c_string, type t:c_ptrConst)
+    where t.eltType == c_char || t.eltType == int(8) || t.eltType == uint(8)
+  {
+    return __primitive("cast", t, x);
+  }
+  //
   // casts from c_ptr(c_char/int(8)/uint(8)) to c_string
   //
   inline operator :(x: c_ptr, type t:c_string)
     where x.eltType == c_char || x.eltType == int(8) || x.eltType == uint(8)
   {
     return __primitive("cast", t, x);
   }
+  //
+  // casts from c_ptrConst(c_char/int(8)/uint(8)) to c_string
+  //
+  inline operator :(x: c_ptrConst, type t:c_string)
+    where x.eltType == c_char || x.eltType == int(8) || x.eltType == uint(8)
```

Comment:
Is there a motivating case for supporting this and the one above for `[u]int(8)` elems? I would slighlty prefer limiting to `c_char` for now as it is the most obvious case. And expanding support as needed. Though I am hoping that we can deprecate `c_string` before then.

Reply:
Nevermind, saw that a test has it. Let's keep this in for symmetry for now.

Reply:
Works for me. I do hope to deprecate c_string soon so I wasn't concerned about making `c_ptrConst` interactions with `c_string` any better than symmetrical to `c_ptr`.